bitswap.go

```diff
@@ -60,7 +58,7 @@ var rebroadcastDelay = delay.Fixed(time.Minute)
 // delegate.
 // Runs until context is cancelled.
 func New(parent context.Context, network bsnet.BitSwapNetwork,
-	bstore blockstore.Blockstore) exchange.Interface {
+	bstore blockstore.Blockstore, provideEnabled bool) exchange.Interface {
```

Comment:
Ideally, we'd use the functional options pattern: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis. Example: https://github.com/libp2p/go-libp2p-kad-dht/blob/master/opts/options.go

Reply:
I don't feel strongly about functional options vs a config param, but I do feel somewhat strongly about not adding as a raw parameter, cause the bitswap initialization params are already long.

Reply:
Honestly though I'm also ok to do this change myself shortly after this is merged if it feels too in the weeds with Bitswap.

Reply:
@hannahhoward thanks, but I don't mind making the change, going to take a look at this today

Reply:
@Stebalien / @hannahhoward check this again and resolve if you're happy with the solution. However, do not merge as there's something going on with the tests and I'm working on it.